collect_app/src/main/java/org/odk/collect/android/widgets/BaseGridWidget.java

```diff
@@ -73,16 +75,26 @@
     List<Integer> selectedItems = new ArrayList<>();
     View[] itemViews;
 
-    public BaseGridWidget(Context context, FormEntryPrompt prompt, boolean quickAdvance, AudioHelper audioHelper) {
-        super(context, prompt, audioHelper);
+    public BaseGridWidget(Context context, QuestionDetails questionDetails, boolean quickAdvance) {
+        super(context, questionDetails);
 
         this.quickAdvance = quickAdvance;
-        noButtonsMode = WidgetAppearanceUtils.isCompactAppearance(prompt) || WidgetAppearanceUtils.isNoButtonsAppearance(prompt);
+        noButtonsMode = WidgetAppearanceUtils.isCompactAppearance(questionDetails.getPrompt()) || WidgetAppearanceUtils.isNoButtonsAppearance(questionDetails.getPrompt());
         itemViews = new View[items.size()];
 
         setUpItems();
         setUpGridView();
         fillInAnswer();
+
+        for (SelectChoice choice : items) {
```

Comment:
How about logging once when the first audio file is found rather than for every choice? I don't think logging for every choice adds any info and there could potentially be a lot of choices.

Media is not supported for grid variants with buttons because there's no layout for putting media, right? I don't see that documented in https://github.com/opendatakit/docs/pull/1113/.

Reply:
Yeah you're right we should only log one. More and more I think it would be nicer to have these events happen before we get to the widgets. Going to have a lil think about that and the your point about `QuestionDetails` and see if there is an easy rework. I do for the moment feel like the `formAnalyticsID` is a detail of the question that the Widgets need but I don't 100% feel like they should need it.